.github/workflows/ci.yml

```diff
@@ -6,6 +6,23 @@ on:
   pull_request:
 
 jobs:
+  type_check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+
+      - name: Install Mypy Dependencies
+        run: pip install mypy
+
+      - name: Type Checking with Mypy
+        run: mypy --strict --install-types --non-interactive learnosity_sdk
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously. -->
**🎯 suggestion:** Let's add the configuration to `pyproject.toml` so it's consistent no matter how it's run. I think we can add the suggested types to `TEST_REQUIRES` as well similar to above

Reply:
Thanks for resolving :+1: